src/patterns.rs

```diff
@@ -259,56 +259,52 @@ fn rewrite_tuple_pat(pats: &[ptr::P<ast::Pat>],
         pat_vec.insert(pos, dotdot);
     }
 
-    if pat_vec.is_empty() {
-        path_str
```

Comment:
The way all the `try_opt`s work in rustfmt is that most rewrite functions return an `Option<String>`. If they successfully rewrite, they return the result in a `Some`, if they cannot rewrite in the space provided, then they return `None`. A small number of functions will try harder by making more space if one of their calless return `None`, but most just fail themselves. So if we are hitting an early return there, it basically means we are formatting at all (rustfmt has a fail-safe mechanism where if we don't format a node, we return the original text - that might be what is happening here. Obviously it is better if we don't do that).